app/src/main/kotlin/org/fossify/gallery/fragments/ViewPagerFragment.kt

```diff
@@ -65,7 +67,19 @@ abstract class ViewPagerFragment : Fragment() {
         }
 
         if (detailsFlag and EXT_RESOLUTION != 0) {
-            context!!.getResolution(file.absolutePath)?.formatAsResolution().let { if (it?.isNotEmpty() == true) details.appendLine(it) }
+            if(medium.name.endsWith(".jxl")) {
+                var jxlCoder = JxlCoder
+                var size = jxlCoder.getSize(file.readBytes())
+                if(size!=null) {
+                    var point = Point(size.width,size.height)
+                    point.formatAsResolution()
+                        .let {if (it.isNotEmpty() == true) details.appendLine(it) }
+                }
+            }else {
+                context!!.getResolution(
+                    file.absolutePath)?.formatAsResolution()
+                    .let { if (it?.isNotEmpty() == true) details.appendLine(it) }
+            }
```

Comment:
This whole block is unformatted. Please format it by selecting it and then running Code ➜ Reformat Code in Android Studio.